### ISSUE TITLE
Provide right click context menu option for users to block 1st party Ads

### ISSUE THREAD

**Issue by srirambv** [brave/browser-laptop#7521]
- Did you search for similar issues before submitting this one?
Yes

- Describe the issue you encountered:
Provide right click context menu option for users to block 1st party Ads

- Platform (Win7, 8, 10? macOS? Linux distro?):
All

- Brave Version (revision SHA):
Brave	0.13.5
rev	1db81cb

- Steps to reproduce:
    1. Visit https://rediff.com
    2. Web page shows 1st party ads which are not blocked
    3. Users should have an option to block 1st party ads
    
- Actual result:
1st party ads are not blocked on sites

- Expected result:
User should have an option to block 1st party ads on websites along with the 3rd party ads that are blocked by default

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes

- Is this an issue in the currently released version?
Yes

- Can this issue be consistently reproduced?
Yes

- Extra QA steps:
    1.
    2.
    3.

- Screenshot if needed:
Highlighted 1st party ads
![image](https://cloud.githubusercontent.com/assets/17010094/23681338/948fab04-0386-11e7-9143-54dc0f0c2dc4.png)

- Any related issues:
cc: @lukemulks

**Comment by lukemulks** [comment]
+1
Would also help with issues where paywall prompts block content until users whitelist the site w/their ad blocker. 

Additional adblock support for blocking inline-scripts would be helpful here too.

**Comment by Jacalz** [comment]
Here is an idea of what i think it should look like: Just right clicking somewhere and pressing "block element". Then you can choose what element to block and it adds it as a filter in a "cosmetic filters" section in settings > shields > Manage Adblock Settings... :lion: Just like in Ublock Origin :+1:

**Comment by Liunkae** [comment]
+1 from #8210

**Comment by pmo** [comment]
I made a post that got closed and added as duplicate and pointed to this one. But this post doesnt mention anything about adding blocklists by url. [https://github.com/brave/browser-laptop/issues/8451](url)

**Comment by Jacalz** [comment]
Oh sorry my bad, missed that part, will open that again 😅

**Comment by luixxiul** [comment]
can this be closed in favor of #344?

**Comment by srirambv** [comment]
#344 is only for cosmetic ads similar to the ones shown on Google after search. This is specifically for 1st party ads which are not blocked.

**Comment by pmo** [comment]
This is what I think should be added, just to be clear.

![asdasdasd](https://cloud.githubusercontent.com/assets/571139/25337452/37eddcfe-28fc-11e7-8d8e-9393150becb5.PNG)

**Comment by Jacalz** [comment]
+1 from https://community.brave.com/t/browser-still-shows-google-ad/2402

**Comment by srirambv** [comment]
+1 from community https://community.brave.com/t/right-mouse-button-menu-add-block-element/5893/2

**Comment by eljuno** [comment]
+1 from community https://community.brave.com/t/right-click-button-on-blocking-images-and-ads-idea/7547?u=eljuno

> right click button on the mouse to block images and ads

**Comment by pongraczi** [comment]
+100000000 thanks :)
maybe including some "trivial big players" individually, like this:  


**Disable 1st party ads:**  
- [ ] Disable globally (everything)  

**Exceptions (enabled, even global disable active):**  

- [ ] Google Search
- [ ] Google Youtube
- [ ] Google Mail
- [ ] Facebook  

**Add your exception domains here:**
```
[_____________________] [  add ]
[ myfavouriteads.com ]  [remove]
[ whateveris.org     ]  [remove]
[ *.specnews.com     ]  [remove]
```

**Comment by BraveSauce** [comment]
The title should change to "manually block content" and not "1st party ads".

On Chrome I use _Ublock Origin_ to block all sort of annoying things that aren't "ads", tailoring the UI of my favorite websites to suit my needs.


Examples of content I block: 

- Resilient cookie reminders that will pop up each and every time, or other kinds of modern popups.

- Annoying "Buy Premium/Upgrade" UI elements on freemium services that I don't wish to pay for.

**Comment by Jacalz** [comment]
+1 on changing the title to something like "add option to right click to remove content"
I am totally agreeing with the user cases that @BraveSauce mentioned above :+1:

**Comment by candrews0** [comment]
+1 this issue. It's a dealbreaker to my using Brave.

**Comment by dri94** [comment]
+1 on this issue

**Comment by steph643** [comment]
Very disappointed to see Google search ads not being blocked.

**Comment by PupoSDC** [comment]
Is this feature being worked on? Without this Brave is not really worth it...

**Comment by AstralGeek33** [comment]
Any updates? I would love to switch to Brave but can't due to the lack of this feature. (As it's nice to be able to do manual blocking.)

It would also be helpful if blocked elements were saved to the cloud so you don't have to re-block the elements upon a new installation.

**Comment by srirambv** [comment]
+1 from https://community.brave.com/t/hover-block-ads/15282/2

**Comment by srirambv** [comment]
+1 from @chris1000  via #14340

**Comment by StephenOwen07** [comment]
+1

**Comment by corysolovewicz** [comment]
+1 
I would definitely utilize this functionality as I frequently run into this type of thing (see attached)
<img width="1123" alt="screen shot 2018-07-03 at 1 58 27 pm" src="https://user-images.githubusercontent.com/19578733/42244280-43fbacd2-7ec9-11e8-8f27-871f874a23ce.png">

**Comment by bsclifton** [comment]
Closing as a duplicate of https://github.com/brave/browser-laptop/issues/344

@Snuupy is working on this for Brave Core 😄 For more info, subscribe to https://github.com/brave/brave-browser/issues/94
